astronomer/providers/snowflake/triggers/snowflake_trigger.py

```diff
@@ -49,14 +46,15 @@ async def run(self) -> AsyncIterator["TriggerEvent"]:  # type: ignore[override]
         Makes a series of connection to snowflake to get the status of the query by async get_query_status function
         """
         hook = get_db_hook(self.snowflake_conn_id)
-        while True:
-            try:
-                run_state = await hook.get_query_status(self.query_ids)
-                if run_state:
-                    yield TriggerEvent(run_state)
-                    return
-                else:
-                    error_message = f"{self.task_id} failed with terminal state: {run_state}"
-                    raise AirflowException(error_message)
-            except AirflowException:
-                await asyncio.sleep(self.polling_period_seconds)
+        try:
```

Comment:
Can you explain this change, please?

Reply:
There was an unnecessary `while` loop removed it and we need to return the TriggerEvent back with error, so instead of `AirflowException` added normal `Exception` and returned the proper TriggerEvent along with error message